rclcpp/include/rclcpp/node_interfaces/node_parameters_interface.hpp

```diff
@@ -105,6 +106,46 @@ class NodeParametersInterface
     rcl_interfaces::msg::ParameterDescriptor(),
     bool ignore_override = false) = 0;
 
+  /// Declare parameters.
+  /**
+   * \sa rclcpp::Node::declare_parameters
+   */
+  RCLCPP_PUBLIC
+  virtual
+  std::vector<rclcpp::ParameterValue>
+  declare_parameters(
+    const std::string & namespace_,
+    const std::vector<rclcpp::Parameter> & parameters,
+    bool ignore_override = false) = 0;
+
+  /// Declare parameters.
+  /**
+   * \sa rclcpp::Node::declare_parameters
+   */
+  RCLCPP_PUBLIC
+  virtual
+  std::vector<rclcpp::ParameterValue>
+  declare_parameters(
+    const std::string & namespace_,
+    const std::vector<
+      std::pair<rclcpp::Parameter, rcl_interfaces::msg::ParameterDescriptor>
+    > & parameters,
+    bool ignore_override = false) = 0;
+
+  /// Declare one or more parameters, all at once.
+  /**
+   * \sa rclcpp::Node::declare_parameters_atomically
+   */
+  RCLCPP_PUBLIC
+  virtual
+  std::vector<rclcpp::ParameterValue>
```

Comment:
this does not seem to be atomic operation since it can return the result for each parameter to declare? `set_parameters_atomically` returns only `rcl_interfaces::msg::SetParametersResult`.

Reply:
same as above.

Reply:
again, this is not about `set_parameters_atomically` but atomic operation to return the `a` result for the operation.

Reply:
But `set_parameters_atomically` returns a `rcl_interfaces::msg::SetParametersResult`, whereas `declare_parameter` only returns `rclcpp::ParameterValue`. `declare_parameter` just throws exceptions if it fails, so I don't think there is a way to distinguish between an atomic version and a non-atomic version unless a new interface is introduced.